packages/kbn-grid-layout/grid/grid_height_smoother.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css } from '@emotion/react';
+import React, { PropsWithChildren, useEffect, useRef } from 'react';
+import { combineLatest } from 'rxjs';
+import { GridLayoutStateManager } from './types';
+
+export const GridHeightSmoother = ({
+  children,
+  gridLayoutStateManager,
+}: PropsWithChildren<{ gridLayoutStateManager: GridLayoutStateManager }>) => {
+  // set the parent div size directly to smooth out height changes.
+  const smoothHeightRef = useRef<HTMLDivElement | null>(null);
+  useEffect(() => {
+    const subscription = combineLatest([
+      gridLayoutStateManager.gridDimensions$,
+      gridLayoutStateManager.interactionEvent$,
+    ]).subscribe(([dimensions, interactionEvent]) => {
+      if (!smoothHeightRef.current) return;
+      if (!interactionEvent) {
+        smoothHeightRef.current.style.height = `${dimensions.height}px`;
+        return;
+      }
+
+      /**
+       * When the user is interacting with an element, the page can grow, but it cannot
+       * shrink. This is to stop a behaviour where the page would scroll up automatically
+       * making the panel shrink or grow unpredictably.
+       */
+      smoothHeightRef.current.style.height = `${Math.max(
+        dimensions.height ?? 0,
+        smoothHeightRef.current.getBoundingClientRect().height
+      )}px`;
+    });
+    return () => subscription.unsubscribe();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div
+      ref={smoothHeightRef}
+      css={css`
+        overflow-anchor: none;
+        transition: height 500ms linear;
```

Comment:
By doing this, we've "accidentally" created a smooth transition when closing a section.... But for some reason it's not applying when opening a section? 🤔 Probably worth looking into at some point.... (not as part of this, just a note for myself)


https://github.com/user-attachments/assets/514ecba4-eca8-4e9b-be0e-b31fb99f8c5e

Reply:
I noticed that and kinda liked it to be honest.

We aren't transitioning when opening the sections because the height smoother doesn't have `overflow: hidden`, so the contents grow and just overflow the smooth parent container. I wasn't able to find a good way to do that, but could be an interesting followup.